python/strings_udf/strings_udf/__init__.py

```diff
@@ -5,32 +5,19 @@
 import subprocess
 import sys
 
+from cubinlinker.patch import _numba_version_ok, get_logger, new_patched_linker
 from numba import cuda
+from numba.cuda.cudadrv.driver import Linker
 from ptxcompiler.patch import CMD
 
 from . import _version
 
 __version__ = _version.get_versions()["version"]
 
-ENABLED = False
+logger = get_logger()
 
-
-def compiler_from_ptx_file(path):
-    """Parse a PTX file header and extract the CUDA version used to compile it.
-    Here is an example PTX header that this function should parse:
-    // Generated by NVIDIA NVVM Compiler
-    //
-    // Compiler Build ID: CL-30672275
-    // Cuda compilation tools, release 11.5, V11.5.119
-    // Based on NVVM 7
-    """
-    file = open(path).read()
-    major, minor = (
-        re.search(r"Cuda compilation tools, release ([0-9\.]+)", file)
-        .group(1)
-        .split(".")
-    )
-    return int(major), int(minor)
+# TODO: embed this in the .so and read dynamically?
```

Comment:
I think we concluded for the fact that this will change infrequently and it will be obvious in CI when it's wrong, that we probably are better off not doing that.